### ISSUE TITLE
Regular 'meltdowns' from server on v5.0.1

### ISSUE THREAD

**Issue by KaylebJay** [minetest/minetest#9251]
Server: `Tunnelers' Abyss` 
Version: `5.0.1 stable`

- Summary

First off, I know this is irrelevant because it does not reflect on the current version of minetest - I'm more trying to understand why this is happening. The server has very regular crashes which take place after ridiculous amounts of 'lag', the players and myself call them meltdowns. The causation is currently unknown, but here are the clues.

1. Meltdowns happen with 16 players. ALWAYS. 
2. Meltdowns make no new mapchunks load. But entities still move around just fine.
3. No player actions are processed, but the server's ping is still fine and the max_lag shown is also incredibly low and normal.
4. The meltdowns will last usually a whole half hour. If an admin is not around to quickly trigger a restart, the players will suffer.
5. Clients stay online in the logs until the whole meltdown is passed, they can logout but they will still display as logged in.
6. The servers.minetest.net display of this is also very strange. Our lag stays almost exactly the same and all clients are shown, through all updates (and ping is fine as already noted).
7. Any dug blocks, client actions registered at *any* time during the meltdown are not processed, unless you stay online the whole time. 

While this seems much like [this issue](https://github.com/minetest/minetest/issues/8584), it also doesn't make much sense for the following reason. Before, I thought that it was just lag from the fact it is a big server, and has lots of heavy mods. But then it didn't make sense... we were fine up to a certain point when the server just popped. So we tried installing LuaJIT, which worked like a charm. The server was absolutely lag-less for a good (almost) two weeks. **No** meltdowns. If it does happen to be fixable by [this nice PR](https://github.com/minetest/minetest/pull/9114), that's great, but why did it not act up after we installed LuaJIT?

Now the meltdowns are at a rather critical state - already almost a dozen over the past week.

So... What exactly **causes** this? Why did it never happen until a certain day of the server's runtime, and now it is a problem? Is it a certain client packet that is causing the issue?

- Related issues
[#9111 ](https://github.com/minetest/minetest/issues/9111)
[#8584 ](https://github.com/minetest/minetest/issues/8584)

I suspect we can add the PR and test how it affects the server :) Some explanation of this all would be very good, though!

**Comment by SmallJoker** [comment]
Testing 5.2.0-dev would be a good idea overall. Maybe it's already fixed by the mentioned #9114 or related PRs.

**Comment by fluxionary** [comment]
@KaylebJay can you post a copy of your minetest.conf file (with anything sensitive removed, if there is anything). I'm wondering if you're unwittingly starving the game of some resource. The "exactly 16 players" thing reminds me of [this](https://github.com/minetest/minetest/blob/44458896d9097c15c722c5bffb3d72e14962b7ce/minetest.conf.example#L1190), though i really have no idea if that particular setting is plausibly responsible for this issue.

**Comment by KaylebJay** [comment]
The server is now updated to 5.2.0-dev and testing will begin. :) @fluxionary If the issue persists it might be that. We'll just have to see...  99% of our minetest.conf is just default settings.

**Comment by KaylebJay** [comment]
So, after a week, here's the results of 5.2.0-dev.
The lag is gone for the most part. For the first few days, the mapgen worked slowly. It seems like 5.2.0 puts more priority on inventory moves, digging, building, and chatting, and the mapgen is called less often.
I was able to fix this by patching a few mods and now the mapgen loads quickly with 20+ players. So overall, it has completely fixed the meltdown problem!
Closing this issue, now that the server is working well :D
